Siva - Restrict past date selection in event search calendar

pavanputti opened the pull request:
# Description
<img width="680" height="450" alt="Screenshot 2025-11-07 at 1 30 47 AM" src="https://github.com/user-attachments/assets/5b467403-bd1f-44a1-8ffa-37313491df2c" />

## Related PRS (if any):
This frontend PR is related to the development backend PR. To test this frontend PR, you need to checkout the development backend PR.
…

## Main changes explained:
- Update `src/components/CommunityPortal/CPDashboard.jsx` to add date restriction functionality
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. Navigate to the Community Portal: /communityportal/.
6. Go to the "All Events" page
7. In the "Search Filters" section on the left, locate the date input field (below "Tomorrow" and "This Weekend" radio buttons)
8.  Click on the date input field to open the calendar picker
9.  **Verify:** Past dates (dates before today) are disabled/grayed out and cannot be selected
10.  **Verify:** Today's date and future dates are selectable and highlighted

## Screenshots or videos of changes:
Before :
<img width="599" height="368" alt="Screenshot 2025-11-07 at 1 31 24 AM" src="https://github.com/user-attachments/assets/a10a0e37-de73-4877-b5a5-bc124e0153cc" />

After 
<img width="1440" height="821" alt="Screenshot 2025-11-07 at 1 24 24 AM" src="https://github.com/user-attachments/assets/fc77ab21-8e30-41f7-b266-06271b90a7bb" />

pavanputti committed:
Restrict past date selection in event search calendar

pavanputti committed:
Merge remote-tracking branch 'origin/development' into siva/fix-past-date-search-issue

Prem203 reviewed (dismissed):
Reviewed the PR with an admin account and the date picker does not allow selection of past dates.

<img width="1911" height="968" alt="image" src="https://github.com/user-attachments/assets/19499a64-dc9d-428e-9741-f7e331a1befb" />

pavanputti committed:
Resolve merge conflicts with development branch - fix past date search issue

pavanputti committed:
Merge development: resolve CPDashboard date conflict (keep min=today), fix duplicate CSS selector

pavanputti committed:
Merge development: resolve conflicts in EquipmentDetailPage.module.css and CPDashboard.jsx, keep module.css pattern and min date fix

pavanputti committed:
Merge development: resolve conflicts in EquipmentDetailPage.module.css and CPDashboard.jsx

pavanputti committed:
Merge development: resolve conflicts in EquipmentDetailPage.module.css and CPDashboard.jsx

HemanthNidamanuru reviewed (changes requested):
Hi,

I tested this PR locally. Past dates are disabled correctly, but I’m not able to select today’s date. Could you please check if this is expected?
<img width="1892" height="1083" alt="PR 4338 - 1" src="https://github.com/user-attachments/assets/5971cc08-4543-464e-a374-b52fae6250a9" />

Anusha-Gali reviewed (changes requested):
Hi Siva,

I have reviewed your PR locally and though past dates are disabled as per requirement. I am unable to select today's date as it too is disabled.

<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/19cf048e-fa53-455b-bedf-63eed1dbf3d3" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/66d26e02-8d53-4688-9431-199b0d5638c7" />

pavanputti committed:
Fix date filter: use local date for min so today is selectable

pavanputti committed:
Merge development: resolve CPDashboard conflict, keep local today and organizer/logo helpers

pavanputti commented:
> Hi,
> 
> I tested this PR locally. Past dates are disabled correctly, but I’m not able to select today’s date. Could you please check if this is expected? <img alt="PR 4338 - 1" width="1892" height="1083" src="https://private-user-images.githubusercontent.com/156004246/560624777-5971cc08-4543-464e-a374-b52fae6250a9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM1MjI0NDIsIm5iZiI6MTc3MzUyMjE0MiwicGF0aCI6Ii8xNTYwMDQyNDYvNTYwNjI0Nzc3LTU5NzFjYzA4LTQ1NDMtNDY0ZS1hMzc0LWI1MmZhZTYyNTBhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMxNFQyMTAyMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00M2E5ZTZkOTNhZjY5NjkxOTMwYmExMmZjZDJiMTFjN2EzMTg5YTg3ZmE0M2NlY2RkYjdhZjkwYjhmMDQxNTQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8v6IC7GHXSFdrfqOfQjZxfjyk0pvh_qnEbikX71BK-c">

Hi @HemanthNidamanuru 

Thanks for pointing it out. I’ve fixed the date logic so today’s date is now selectable.

pavanputti committed:
Fix CPDashboard line length for prettier/lint

Anusha-Gali reviewed (dismissed):
Hi Siva,

Thanks for fixing the issue, it works as per requirement now.

<img width="1274" height="829" alt="Image" src="https://github.com/user-attachments/assets/4ac8dead-bd40-4211-ace3-98ab38a910cd" />

Vishnu10-hub reviewed (changes requested):
Hi,

I ran the PR locally. The past dates are correctly disabled, which works as expected. However, for today’s date, no data is showing in the calendar/events.


<img width="1440" height="882" alt="Image" src="https://github.com/user-attachments/assets/3e5f6039-1eab-4a31-9e50-71554c22719d" />
<img width="1440" height="882" alt="Image" src="https://github.com/user-attachments/assets/d9271cd0-734d-4db3-b433-79953a55cfe1" />

pavanputti committed:
Phase 3 - Restrict Selection of Past Dates in Calendar

- Add minDate={new Date()} to CommunityCalendar ReactCalendar to disable past date selection
- Add min attribute (today) to CPDashboard date filter input; wire showPicker() so clicking the chevron opens the calendar
- Add min attribute (today) to ActivityList date filter input
- Add min attribute (today) to RegistrationForm event date input
- Add min attribute (today) to CreateEventModal event date input
- Convert CommentSection.css to CommentSection.module.css and update classNames
- Convert reportsPage.css to reportsPage.module.css; update Reports.jsx, ReportTableSearchPanel.jsx, and AddLostTime.jsx to use CSS Modules

pavanputti committed:
Resolve merge conflicts with development

pavanputti committed:
Fix duplicate prop in CreateEventModal.jsx

pavanputti committed:
Remove unused variables today and handleDateInputClick

pavanputti committed:
Fix prettier formatting in CPDashboard.jsx

one-community commented:
Thank you all, merging!
